packages/block/src/block.ts

```diff
@@ -93,17 +95,22 @@ export class Block {
     header: BlockHeader,
     transactions: Transaction[],
     uncleHeaders: BlockHeader[],
-    //data: Buffer | [Buffer[], Buffer[], Buffer[]] | BlockData = {},
     opts: BlockOptions = {},
   ) {
     this.header = header
     this.transactions = transactions
     this.uncleHeaders = uncleHeaders
     this._common = this.header._common
+
+    Object.freeze(this)
```

Comment:
I'll have to give it a test, but I think the common object can still update, it just can't be set to a totally new instance (e.g. `block._common = newCommon` would fail)

Reply:
Yes I think this is the case according to here:

> Note that values that are objects can still be modified, unless they are also frozen.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/freeze